sentry-contrib-native-sys/build.rs

```diff
@@ -83,12 +105,35 @@ fn build(out_dir: &Path, source: &Path, install: &Path) -> Result<()> {
         .context("failed to parse path")?
         .to_owned();
 
-    if !Command::new("cmake")
-        .current_dir(source)
-        .args(&["-B", &build, "-D", "BUILD_SHARED_LIBS=OFF"])
-        .status()?
-        .success()
+    let mut cfg_cmd = Command::new("cmake");
+    // Build static libraries
+    cfg_cmd.args(&[
+        "-B",
+        &build,
+        "-D",
+        "BUILD_SHARED_LIBS=OFF",
+        "-D",
+        "SENTRY_BUILD_TESTS=OFF",
+        "-D",
+        "SENTRY_BUILD_EXAMPLES=OFF",
+    ]);
+
+    // Apparently cmake defaults to windows 32 bits? wtf
```

Comment:
that is interesting. Might be worth documenting better in sentry-native how to use this properly.

Reply:
This may be because the build script is directly calling cmake, I would assume the cmake crate would properly set these kind of things based upon the target triple, but this was (so far) the only cmake-ism that I've encountered. My long term goal would be to get rid of cmake entirely, but since there are so many C++ dependencies maybe that's a fight not worth having, at least for now.

Reply:
I've encountered this many time, but apparently updating to the newest CMake and Visual Studio version seems to fix this, I'm unaware of the details.
But I would prefer not trying to fix CMake in this crate, can you see if updating fixes your issues? If not maybe we should rely on https://crates.io/crates/cmake instead?

Reply:
I have CMake 3.17.2 on a windows machine, but VS2017 build tools, though it would be strange to me for the version of VS to affect this, but then again, it wouldn't surprise me I suppose. 

The CMake crate does seem to be doing this [correctly](https://github.com/alexcrichton/cmake-rs/blob/master/src/lib.rs#L460-L481), so might be worth checking out.

Reply:
Oh, and they helpfully pointed out that it is the VS version which affects that.

> // 32-bit x86 toolset used to be the default for all hosts,
> // but Visual Studio 2019 changed the default toolset to match the host,
> // so we need to manually override it for x86 targets

Reply:
I'm going to look into using the `cmake` crate.

Reply:
I replaced this whole thing with the `cmake` create now in https://github.com/daxpedda/sentry-contrib-native/commit/1bbf83ed7b561fad82d05eb07bbdcf71c0f83589.